src/renderer/containers/App/handleUploadJobUpdates.ts

```diff
@@ -0,0 +1,115 @@
+import { Dispatch } from "react";
+
+import { FSSUpload, UploadStatus } from "../../services/file-storage-service";
+import {
+  JSSJob,
+  Service,
+  UploadJob,
+} from "../../services/job-status-service/types";
+import {
+  receiveFSSJobCompletionUpdate,
+  receiveJobUpdate,
+  updateUploadProgressInfo,
+} from "../../state/job/actions";
+import { Step } from "../Table/CustomCells/StatusCell/Step";
+
+/**
+ * Handles job updates for a standard file upload.
+ * Reports progress on the pre-upload MD5 step, the upload step, and then the post-upload MD5 step.
+ */
+function handleFSSJobUpdate(job: FSSUpload, dispatch: Dispatch<any>) {
+  const totalBytes = job.serviceFields.fileSize || 0; // 0 is a safe default, but in practice filesize is initialized immediately after job creation.
+  if (
+    job.serviceFields?.preUploadMd5 &&
+    job.serviceFields.preUploadMd5 !== job.serviceFields.fileSize
```

Comment:
Why would the preUploadMd5 ever equal the fileSize?  When they are both default initialized to zero or something?

Reply:
This is copy-pasted / moved code that is certainly not "self-documenting"... so I'm not sure.

See line 28 just below this where the value for `bytesUploaded` is the `preUploadMd5` value. Can't say I have any clue what that actually means. I would assume `preUploadMd5` also starts at 0, just like how `totalbytes` is defaulted to on line 21. So you may be right about this being a check for... reasons, at the start of each step, but again I'm not sure.

Reply:
Could warrant some investigation just for better understanding. Another reason why some kind of documentation re: how we use JSS in here could be useful